### ISSUE TITLE
proxyconnect tcp: dial tcp 192.168.64.1:3128: connect: connection refused [with experimental features enabled]

### ISSUE THREAD

**Issue by alphavector** [docker/for-mac#5663]
<!--

************************************************************************************
*READ ME FIRST*

This repository is for reporting bugs with the Docker Desktop for Mac software,
which we respond to on a best-effort basis.

Support requests in this repository (i.e., trouble installing or using the
software) will be ignored, but personalized support is available to Docker Pro and
Team customers at https://hub.docker.com/support/desktop/, and community support is
available from the Docker community Slack (https://www.docker.com/docker-community).

Issues without a clear title, description, and diagnostics ID cannot be debugged,
and will be closed. See https://docs.docker.com/docker-for-mac/troubleshoot/#diagnose-and-feedback
for how to generate a diagnostics ID either from inside the app or from the command
line.
************************************************************************************

-->


<!-- Click these checkboxes after submitting, or place an "x" in them. -->
  - [x] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: FDFB80A8-65E5-4629-A5C5-DB767BFD0508/20210507155721

### Expected behavior
Creating an image

### Actual behavior
```
[+] Building 0.8s (3/3) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                       0.4s
 => => transferring dockerfile: 90B                                                                                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                          0.5s
 => => transferring context: 2B                                                                                                                                                                                                                                            0.0s
 => ERROR [internal] load metadata for docker.io/library/debian:buster-slim                                                                                                                                                                                                0.0s
------
 > [internal] load metadata for docker.io/library/debian:buster-slim:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: failed to do request: Head https://registry-1.docker.io/v2/library/debian/manifests/buster-slim: proxyconnect tcp: dial tcp 192.168.64.1:3128: connect: connection refused
```

### Information
<!--
Please, help us understand the problem.  For instance:
  - Is it reproducible?
  - Is the problem new?
  - Did the problem appear with an update?
  - A reproducible case if this is a bug, Dockerfiles FTW.
-->
The problem occurs **only** when the "Use new virtualization framework" option is enabled
  - macOS Version: 11.3.1 (20E241)
  - Intel chip or Apple chip: Apple chip M1
  - Docker Desktop Version: 3.3.3 (64133)

### Steps to reproduce the behavior
<!--
A reproducible case, Dockerfiles FTW.
-->

  1. Create a Dockerfile with content:
  ```
  FROM debian:buster-slim
RUN echo "Hello, World!"
  ```
  2. Run `docker build .`

**Comment by linkinkov** [comment]
I can confirm its too, in 3.3.1 experimental virtualization work well

**Comment by YanZzz** [comment]
Today I just upgrade to latest version 3.3.3. after that everything went to  bad.
**~ % docker pull hello-world**
Using default tag: latest
Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

In order to bring it back to working state, I have to disable the "Use new virtualization framework" (uncheck it.)

then it works.
 ~ % docker pull hello-world
Using default tag: latest
latest: Pulling from library/hello-world
256ab8fe8778: Pull complete 
Digest: sha256:f2266cbfc127c960fd30e76b7c792dc23b588c0db76233517e1891a4e357d519
Status: Downloaded newer image for hello-world:latest
docker.io/library/hello-world:latest

**Comment by alphavector** [comment]
I tested it on experimental build 64397 -- no problem, the container is pulled  
* Apple Silicon: https://desktop-stage.docker.com/mac/stable/arm64/64397/Docker.dmg
* Intel: https://desktop-stage.docker.com/mac/stable/amd64/64397/Docker.dmg

@djs55, thanks for the fix, awaiting fix in stable build

**Comment by joschi** [comment]
Same problem here with Intel-based Macbook Pro 16".

- macOS Version: 11.3.1 (20E241)
- Intel chip or Apple chip: Intel
- Docker Desktop Version: 3.3.3 (64133)

**Comment by g1r0** [comment]
Hi, I've got the same issue today with Big Sur virtualization.framework option in use. Falling back to hypervisor.framework made the docker working again.
```
BigSur 11.4
Docker 3.3.3(64133)
```

**Comment by djs55** [comment]
@alphavector the fix from experimental build 64397 has now been released in Docker Desktop 3.4.0. I'll close this now but ping me here if you believe the issue is still present. Thanks for the report!
